nuocd/nuodb_adminquery.py

```diff
@@ -41,37 +41,42 @@ def get_admin_conn(pid):
         except IOError:
             pass
 
+        protocol = None
+        address = None
         api_server = env.get('NUOCMD_API_SERVER', os.environ.get('NUOCMD_API_SERVER'))
-        if api_server is None:
-            # look in /etc/nuodb/nuoadmin.conf to see if ssl is enabled.
-            pass
-
         match = re.match('(https?://|)([^:]+)(:[0-9]+|)', api_server)
         if match:
             protocol = match.group(1)
-            if len(protocol) == 0:
-                protocol = 'https://'
             adminhost = match.group(2)
             port = match.group(3)
             if len(port) == 0:
                 port = ':8888'
-            api_server = protocol + adminhost + port
-
-        if protocol == 'https://':
-            client_key = env.get('NUOCMD_CLIENT_KEY', os.environ.get('NUOCMD_CLIENT_KEY'))
-            if client_key:
-                client_key = ROOT + client_key
-
-            # this assumes set to a path. It could be set to True in which case we need
-            # to return the default path
-
-            server_cert = env.get('NUOCMD_VERIFY_SERVER', os.environ.get('NUOCMD_VERIFY_SERVER', False))
-            if server_cert and server_cert != True:
-                server_cert = ROOT + server_cert
-        else:
+            address = adminhost + port
+
+        client_key = env.get('NUOCMD_CLIENT_KEY', os.environ.get('NUOCMD_CLIENT_KEY'))
+        if client_key:
+            client_key = ROOT + client_key
+            # downgrade if protocol is unspecified and client key does not exist
+            if not protocol and not os.path.exists(client_key):
+                client_key = None
+
+        # this assumes set to a path. It could be set to True in which case we need
+        # to return the default path
+
+        server_cert = env.get('NUOCMD_VERIFY_SERVER', os.environ.get('NUOCMD_VERIFY_SERVER', False))
```

Comment:
server_cert can also be missing. In 4.2 we no longer ship either the server cert or the client cert.

A missing server_cert file is "OK" in python?

Reply:
I don't think we should downgrade if the client key exists. If someone went through the trouble to create `nuocmd.pem`, but they forgot to create or misplaced (e.g. they called it `/etc/nuodb/keys/ca.crt` instead of `/etc/nuodb/keys/ca.cert`) the server certificate, then they would probably be expecting HTTPS to be used.